lib/AST/DeclContext.cpp

```diff
@@ -91,6 +91,16 @@ ProtocolDecl *DeclContext::getAsProtocolExtensionContext() const {
 GenericTypeParamType *DeclContext::getProtocolSelfType() const {
   assert(getAsProtocolOrProtocolExtensionContext() && "not a protocol");
 
+  auto genericParams = getGenericParamsOfContext();
```

Comment:
ProtocolDecl::createGenericParamsIfMissing() is lighter weight

Reply:
Indeed, thank you!